reactioncommerce/api-service-account
# Check if groupIds not empty before calling updateGroupsForAccounts

Resolves #27
Impact: **major**
Type: **bugfix**

## Issue
`updateGroupsForAccounts` can throw an error if no groups are provided, and in `createAccount`, the mutation is called regardless of whether there are groups to add or not.

## Solution
Check if there are groups to add to the account in `createAccount` before calling `updateGroupsForAccounts`. This will prevent the mutation from being called when it isn't needed. Also check if `groupIds` were passed before throwing an error about groups not being found.

## Breaking changes
None.

## Testing
1. Create a customer account, expect no errors.
2. Create an admin account from an invitation, expect no errors. 
3. Check that the relevant groups were added to the admin account in the `Accounts` MongoDB collection.
